pilot/pkg/model/push_context.go

```diff
@@ -419,6 +386,38 @@ func (ps *PushContext) VirtualServices(proxy *Proxy, gateways map[string]bool) [
 	return out
 }
 
+// GetSidecarScope returns a sidecar rule applicable to the config
+// namespace associated with the proxy
+func (ps *PushContext) GetSidecarScope(proxy *Proxy, proxyInstances []*ServiceInstance) *SidecarScope {
+
+	var workloadLabels LabelsCollection
+	for _, w := range proxyInstances {
```

Comment:
This seems incorrect - shouldn't it use the labels in the proxy (the actual labels that envoy is passing) instead if the labels from the ServiecInstance ? Are they the same ? I don't think so.

Reply:
(maybe just add a comment and TODO - we can fix it in separate PR, but need to keep track)

Reply:
Maybe also a flag (or env) to only do 'default' Sidecar. 

I'm still not convinced we have time to properly test multiple selectors, and it increases the risks on 1.1

Reply:
This is the exact same logic that we have been using for the last 8 months on gateways and EnvoyFilters. The labels in the proxy instances are the same as the ones that we add via sidecar injector to the proxy metadata. In fact, the proxy instance labels are more upto date compared to the ones on the proxy metadata, as adding/removing a workload label does not get reflected in the proxy metadata.